Build fails on undefined `uint32_t` in protobuf

Here is what I get after doing a `bazel clean --expunge`:

```
➜  verible git:(master) bazel build -c opt //...
INFO: Analyzed 1898 targets (0 packages loaded, 0 targets configured).
INFO: Found 1898 targets...
ERROR: /home/beckmann/.cache/bazel/_bazel_beckmann/d932d629abbf7de2514b0932259df566/external/com_google_protobuf/src/google/protobuf/compiler/BUILD.bazel:73:11: Compiling src/google/protobuf/compiler/subprocess.cc failed: (Exit 1): gcc failed: error executing command /usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer -g0 -O2 '-D_FORTIFY_SOURCE=1' -DNDEBUG -ffunction-sections ... (remaining 63 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
In file included from bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/compiler/_virtual_includes/command_line_interface/google/protobuf/compiler/subprocess.h:47,
                 from external/com_google_protobuf/src/google/protobuf/compiler/subprocess.cc:33:
bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/port.h:186:67: error: 'uint32_t' was not declared in this scope
  186 |     : absl::disjunction<std::is_same<T, int32_t>, std::is_same<T, uint32_t>,
      |                                                                   ^~~~~~~~
bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/port.h:56:1: note: 'uint32_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
   55 | #include "google/protobuf/port_def.inc"
  +++ |+#include <cstdint>
   56 | 
bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/port.h:186:75: error: template argument 2 is invalid
  186 |     : absl::disjunction<std::is_same<T, int32_t>, std::is_same<T, uint32_t>,
      |                                                                           ^
bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/port.h:187:67: error: 'uint64_t' was not declared in this scope
  187 |                         std::is_same<T, int64_t>, std::is_same<T, uint64_t>,
      |                                                                   ^~~~~~~~
bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/port.h:187:67: note: 'uint64_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/port.h:187:75: error: template argument 2 is invalid
  187 |                         std::is_same<T, int64_t>, std::is_same<T, uint64_t>,
      |                                                                           ^
bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/port.h:188:45: error: template argument 2 is invalid
  188 |                         std::is_same<T, bool>> {};
      |                                             ^~
bazel-out/k8-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/src/google/protobuf/_virtual_includes/protobuf_lite/google/protobuf/port.h:188:45: error: template argument 4 is invalid
ERROR: /home/beckmann/.cache/bazel/_bazel_beckmann/d932d629abbf7de2514b0932259df566/external/com_google_protobuf/src/google/protobuf/BUILD.bazel:142:11 Middleman _middlemen/@com_Ugoogle_Uprotobuf_S_Ssrc_Sgoogle_Sprotobuf_Cwkt_Ucc_Uproto-BazelCppSemantics_build_arch_k8-opt-exec-2B5CBBC6 failed: (Exit 1): gcc failed: error executing command /usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer -g0 -O2 '-D_FORTIFY_SOURCE=1' -DNDEBUG -ffunction-sections ... (remaining 63 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
INFO: Elapsed time: 2.004s, Critical Path: 1.84s
INFO: 67 processes: 57 internal, 10 linux-sandbox.
FAILED: Build did NOT complete successfully
```

The `uint32_t` error is gone, thanks, but now I am seeing this error:

```
➜  verible git:(master) bazel build -c opt //...
INFO: Analyzed 1898 targets (0 packages loaded, 0 targets configured).
INFO: Found 1898 targets...
ERROR: /home/beckmann/.cache/bazel/_bazel_beckmann/d932d629abbf7de2514b0932259df566/external/com_google_boringssl/BUILD:152:11: Compiling src/crypto/bytestring/cbb.c failed: (Exit 1): gcc failed: error executing command /usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer -g0 -O2 '-D_FORTIFY_SOURCE=1' -DNDEBUG -ffunction-sections ... (remaining 36 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
In function 'CBB_add_asn1',
    inlined from 'CBB_add_asn1_octet_string' at external/com_google_boringssl/src/crypto/bytestring/cbb.c:567:8:
external/com_google_boringssl/src/crypto/bytestring/cbb.c:388:14: error: storing the address of local variable 'child' in '*cbb.child' [-Werror=dangling-pointer=]
  388 |   cbb->child = out_contents;
      |   ~~~~~~~~~~~^~~~~~~~~~~~~~
external/com_google_boringssl/src/crypto/bytestring/cbb.c: In function 'CBB_add_asn1_octet_string':
external/com_google_boringssl/src/crypto/bytestring/cbb.c:566:7: note: 'child' declared here
  566 |   CBB child;
      |       ^~~~~
external/com_google_boringssl/src/crypto/bytestring/cbb.c:565:36: note: 'cbb' declared here
  565 | int CBB_add_asn1_octet_string(CBB *cbb, const uint8_t *data, size_t data_len) {
      |                               ~~~~~^~~
In function 'CBB_add_asn1',
    inlined from 'CBB_add_asn1_bool' at external/com_google_boringssl/src/crypto/bytestring/cbb.c:578:8:
external/com_google_boringssl/src/crypto/bytestring/cbb.c:388:14: error: storing the address of local variable 'child' in '*cbb.child' [-Werror=dangling-pointer=]
  388 |   cbb->child = out_contents;
      |   ~~~~~~~~~~~^~~~~~~~~~~~~~
external/com_google_boringssl/src/crypto/bytestring/cbb.c: In function 'CBB_add_asn1_bool':
external/com_google_boringssl/src/crypto/bytestring/cbb.c:577:7: note: 'child' declared here
  577 |   CBB child;
      |       ^~~~~
external/com_google_boringssl/src/crypto/bytestring/cbb.c:576:28: note: 'cbb' declared here
  576 | int CBB_add_asn1_bool(CBB *cbb, int value) {
      |                       ~~~~~^~~
cc1: all warnings being treated as errors
INFO: Elapsed time: 3.053s, Critical Path: 2.81s
INFO: 81 processes: 51 internal, 30 linux-sandbox.
FAILED: Build did NOT complete successfully
```

This is strange because I have installed verible from source several times on this machine recently, so I don't know why I'm running into these issues now.

And I'm on OpenSUSE:

```
➜  verible git:(master) cat /etc/os-release 
NAME="openSUSE Tumbleweed"
# VERSION="20230616"
```

--- a/WORKSPACE
+++ b/WORKSPACE
@@ -230,12 +230,3 @@ maybe(
         "https://zlib.net/fossils/zlib-1.2.13.tar.gz",
     ],
 )
-
-# Only needed for sha256; Version taken from branch with bazel rules
-# https://github.com/google/boringssl/tree/master-with-bazel
-http_archive(
-    name = "com_google_boringssl",
-    sha256 = "8b28b030652d1b5e0f8de306e8bdc1b67f7164c2ec889056a0c13e78ba615a34",
-    strip_prefix = "boringssl-342e805bc1f5dfdd650e3f031686d6c939b095d9",
-    urls = ["https://github.com/google/boringssl/archive/342e805bc1f5dfdd650e3f031686d6c939b095d9.zip"],
-)

--- a/verilog/tools/kythe/BUILD
+++ b/verilog/tools/kythe/BUILD
@@ -277,9 +277,9 @@ cc_library(
         "//common/util:file-util",
         "//common/util:simple-zip",
         "//third_party/proto/kythe:analysis_cc_proto",
+        "@com_google_absl//absl/hash",
         "@com_google_absl//absl/status",
         "@com_google_absl//absl/strings",
-        "@com_google_boringssl//:crypto",
     ],
 )
 

--- a/verilog/tools/kythe/kzip_creator.cc
+++ b/verilog/tools/kythe/kzip_creator.cc
@@ -19,12 +19,12 @@
 #include <string>
 #include <vector>
 
+#include "absl/hash/hash.h"
 #include "absl/status/status.h"
 #include "absl/strings/escaping.h"
 #include "absl/strings/string_view.h"
 #include "common/util/file_util.h"
 #include "common/util/simple_zip.h"
-#include "openssl/sha.h"
 #include "third_party/proto/kythe/analysis.pb.h"
 
 namespace verilog {
@@ -34,12 +34,8 @@ namespace {
 constexpr absl::string_view kProtoUnitRoot = "root/pbunits/";
 constexpr absl::string_view kFileRoot = "root/files/";
 
-std::string SHA256Digest(absl::string_view content) {
-  std::array<unsigned char, SHA256_DIGEST_LENGTH> buf;
-  ::SHA256(reinterpret_cast<const unsigned char *>(content.data()),
-           content.size(), buf.data());
-  return absl::BytesToHexString(absl::string_view(
-      reinterpret_cast<const char *>(buf.data()), buf.size()));
+std::string Digest(absl::string_view content) {
+  return absl::StrCat(absl::HashOf(content));
 }
 
 constexpr int kKZipCompressionLevel = 9;
@@ -58,7 +54,7 @@ KzipCreator::KzipCreator(absl::string_view output_path)
 
 std::string KzipCreator::AddSourceFile(absl::string_view path,
                                        absl::string_view content) {
-  std::string digest = SHA256Digest(content);
+  std::string digest = Digest(content);
   const std::string archive_path = verible::file::JoinPath(kFileRoot, digest);
   archive_.AddFile(archive_path, verible::zip::MemoryByteSource(content));
   return digest;
@@ -70,7 +66,7 @@ absl::Status KzipCreator::AddCompilationUnit(
   if (!unit.SerializeToString(&content)) {
     return absl::InternalError("Failed to serialize the compilation unit");
   }
-  const std::string digest = SHA256Digest(content);
+  const std::string digest = Digest(content);
   const std::string archive_path =
       verible::file::JoinPath(kProtoUnitRoot, digest);
   archive_.AddFile(archive_path, verible::zip::MemoryByteSource(content));

--- a/verilog/tools/kythe/kzip_creator.h
+++ b/verilog/tools/kythe/kzip_creator.h
@@ -34,7 +34,7 @@ class KzipCreator final {
   // Initializes the archive. Crashes if initialization fails.
   explicit KzipCreator(absl::string_view output_path);
 
-  // Adds source code file to the Kzip. Returns its SHA digest.
+  // Adds source code file to the Kzip. Returns its digest.
   std::string AddSourceFile(absl::string_view path, absl::string_view content);
 
   // Adds compilation unit to the Kzip.

